x/distribution/handler.go

```diff
@@ -104,6 +104,8 @@ func NewCommunityPoolSpendProposalHandler(k Keeper) govtypes.Handler {
 	return func(ctx sdk.Context, content govtypes.Content) error {
 		switch c := content.(type) {
 		case types.CommunityPoolSpendProposal:
+			return keeper.HandleCommunityPoolSpendProposal(ctx, k, &c)
+		case *types.CommunityPoolSpendProposal:
 			return keeper.HandleCommunityPoolSpendProposal(ctx, k, c)
```

Comment:
@alexanderbez trying to figure out the best way to deal with pointer proposal instances. By default protobuf generates the std `Content` oneof to hold pointer instances of each of the proposal types. So we need a case to handle pointer types here. Many of our existing tests also use the non-pointer version. My solution was to add cases for both pointer and non-pointer, but this is a little bit cumbersome.

Alternatives are:
* don't have a switch case for the non-pointer version and always expect the pointer version
* try using the gogo proto face plugin to generate an interface and switch on that instead: https://godoc.org/github.com/gogo/protobuf/plugin/face

Reply:
My inclination is to do `don't have a switch case for the non-pointer version and always expect the pointer version` as it's more explicit.

I'm not familiar with the `Face` plugin. How would that work? Would it make the UX nicer? Would it increase surface area for bugs or decrease surface area?

Reply:
The face plugin is explained pretty well in the link I shared above. The main advantage I think would be just being able to switch on one `CommunitySpendProposalFace` type that should work for both pointer and non-pointer versions.

I'm trying to think of what's the best UX for module authors. Module authors probably would assume that the non-pointer version would work and wire things up like that. Then they would test and see that it didn't work and need to switch to the pointer version. The challenge is that zones that try to continue using amino (or codon) would probably always send the non-pointer version so then they would have problems. So if module authors knew to use face that would probably reduce the surface-area for bugs across all zones although it adds to the number of things they need to know.

Reply:
Although maybe the concern for amino users can be addressed by registering explicitly the pointer version, i.e.:
```go
cdc.RegisterConcrete(&SoftwareUpgradeProposal{}, "cosmos-sdk/SoftwareUpgradeProposal", nil)
```

I do want to note that if we are intending to maintain an amino compatible path, it will probably more and more need to be something we have explicit tests for

Reply:
Right, I'd rather enforce clear semantics of the types. If we expect a pointer in most places, we should only allow a pointer. i.e. let's go with option 1.

Reply:
Okay I updates this so that the pointer versions of `gov.Content` instances are always used. It does feel a bit awkward I have to say, but maybe there isn't a better alternative.

Reply:
The point being is that we should make sure the code is clear what semantics we're trying to enforce and that's pointer semantics.